src/util/shell_helpers.rs

```diff
@@ -105,3 +107,43 @@ pub fn run_bash_command_async(cmd: &str, dir: Option<&str>) -> Result<async_proc
         .spawn()
         .context("Failed to spawn bash command")
 }
+
+pub fn launch_command_with_stdout_label(
+    c: &mut Command,
+    attempted_command_name_summary: Option<String>,
+) -> Result<std::process::Child> {
+    // Make sure to pipe stderr/out up to this process for logging.
+    c.stdout(Stdio::piped()).stderr(Stdio::piped());
+
+    // extract the executable name from the command.
+    let executable = attempted_command_name_summary.unwrap_or(
+        c.get_program()
+            .to_str()
+            .context("Failed to get program")?
+            .to_string(),
+    );
+
+    // Spawn the process.
+    let mut child = c
+        .spawn()
+        .context(format!("Failed to execute {executable}"))?;
+
+    // Read in the stdout of the process, prepend it, and forward it to our logging.
+    if let Some(ref mut stdout) = child.stdout {
+        for line in std::io::BufReader::new(stdout).lines() {
+            info!("Subprocess {executable} stdout: {}", line.unwrap());
+        }
+    }
+
+    // Read in the stderr of the process, prepend it with an error, and forward it
+    // to our logging.
+    if let Some(ref mut stderr) = child.stderr {
+        for line in std::io::BufReader::new(stderr).lines() {
```

Comment:
Can you consolidate the above two loops in one so that `stdout` and `stderr` are read in the same loop? This fixes the order of the output and the error messages and avoid blocking child process on the `stderr` pipe.

Reply:
great point, done and added itertools to make it prettier